### ISSUE TITLE
jsx-sort-props: Bug: Sort order is case sensitive

### ISSUE THREAD

**Issue by gasi** [yannickcr/eslint-plugin-react#404]
One could argue feature or bug but for me it’s definitely a bug. The rule `jsx-sort-props` considers the following out of order:

```js
export default class Foo extends React.Component {
  static propTypes = {
    aa: …,
    aB: …,
  }
  // …
}
```

This is because order is determined using `<` operator on strings which uses ASCII order and uppercase letters come before lowercase ones, e.g.

```sh
> node
> 'aa' < 'aB'
false
```

**Source:** https://github.com/yannickcr/eslint-plugin-react/blob/5c1f029e4cd1a03193e3dcbf0e5020986b3232ae/lib/rules/jsx-sort-props.js#L63-L66

I’d have to try, but I believe adding `.toLowerCase()` should get us one step ahead. Then again, doing it properly for international use cases is a bit more involved: https://marcoscaceres.github.io/jsi18n/#localze_numbers :smile: 

Thanks for your work on this plugin — it’s been such a step forward over linting React while using CoffeeScript :bow: :heart:

**Comment by gscottolson** [comment]
:+1:

**Comment by wyze** [comment]
There is an `ignoreCase` option you can pass to the rule.

**Comment by gasi** [comment]
Thanks, we’ll try that. Any reason not to make it the default?

On Saturday, February 20, 2016, Neil Kistner <notifications@github.com>
wrote:

> There is an ignoreCase option you can pass to the rule.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yannickcr/eslint-plugin-react/issues/404#issuecomment-186723618>
> .
>


-- 
Sent from my iPhone

**Comment by ljharb** [comment]
@gasi I'd say because case sensitive sorting is the default in most computer systems, and i'd be very surprised to find `A` sorted before `b`.

**Comment by gasi** [comment]
@ljharb You’re right in the sense that `ls` works like that:

```bash
drwxr-xr-x     2 dani  staff         68 Feb 23 15:01 alphaBeta
drwxr-xr-x     2 dani  staff         68 Feb 23 15:01 alphaalpha
```

Still, for `props`, I find it odd. Glad I can customize it for my own use case :smile:

**Comment by lencioni** [comment]
Considering there is an option to accomplish what you want here, I think I'm going to close this issue out. Please re-open if there is anything else that needs to be done.
